### PR TITLE
[REF] [PHP8.2]Treat subType as a internal varible, not a form property

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Treat subType as a internal varible, not a form property

Before
----------------------------------------
->_subType not used from outside the function

image

After
----------------------------------------
>_subType is an internal property

The only refences left are in `CustomDataByType` - which can also go but would conflict with https://github.com/civicrm/civicrm-core/pull/26832/files#diff-1a220e071c24dde6c43cc3e60d31805da72604d8774dbb7c26d6111a324c9cb4R51

image

Technical Details
----------------------------------------
I also added the deprecation we had in https://github.com/civicrm/civicrm-core/pull/26832/files#diff-1a220e071c24dde6c43cc3e60d31805da72604d8774dbb7c26d6111a324c9cb4L51 so we can phase out that code

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Contact/Form/Edit/CustomData.php [hunk 1]
```diff
@@ -37,7 +36,7 @@ public static function preProcess(&$form) {
       $form->assign("blockName", $form->_addBlockName);
     }
 
-    CRM_Custom_Form_CustomData::preProcess($form, NULL, $form->_subType, NULL,
+    CRM_Custom_Form_CustomData::preProcess($form, NULL, NULL, NULL,
```
Comment (colemanw) [inline CRM/Contact/Form/Edit/CustomData.php:39]: Why is this changed to NULL?

Comment (eileenmcnaughton) [reply CRM/Contact/Form/Edit/CustomData.php:39]: because the `preProcess` function retrieves it from the url with `$form->_subType = CRM_Utils_Request::retrieve('subType', 'String');` if it is NULL

Comment (eileenmcnaughton) [reply CRM/Contact/Form/Edit/CustomData.php:39]: see https://github.com/civicrm/civicrm-core/pull/26833/files#diff-931e65674ed680adf93d7ba4727419ec32b5fd4636ac84109dd9621f89e20e0dR91

Comment (colemanw) [reply CRM/Contact/Form/Edit/CustomData.php:39]: Ah ok.
